pkg/apis/kubeone/types.go

```diff
@@ -48,6 +48,8 @@ type KubeOneCluster struct {
 	MachineController *MachineControllerConfig `json:"machineController,omitempty"`
 	// Features enables and configures additional cluster features
 	Features Features `json:"features,omitempty"`
+	// SystemPackages configure kubeone behaviour regarding OS packages
+	SystemPackages SystemPackages `json:"systemPackages"`
```

Comment:
I'd make this a pointer rather than having a bool pointer. From the usability side, I think it's much better approach.

Reply:
Also for consistency, because we use that approach for machine-controller config too.

Reply:
Then I wouldn't be able to just
```go
"CONFIGUREREPOSITORIES": s.Cluster.SystemPackages.ConfigureRepositories,
```

I'd need to check if `s.Cluster.SystemPackages == nil`

Reply:
But what if `s.Cluster.SystemPackages.ConfigureRepositories` is `nil`, wouldn't it NPE? Also, I think UX is more important than an additional check. :wink:

Reply:
Also, we have defaulting and validation which would ensure that it isn't `nil`.